sound/soc/sof/topology.c

```diff
@@ -2335,6 +2345,15 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	if (!spcm->pcm.capture)
 		return ret;
 
+	ret = sof_parse_tokens(scomp, &spcm->stream[stream], stream_tokens,
```

Comment:
@keyonjie is this second call necessary? The first call to parse stream_tokens will have already parsed both the PLAYBACK and CAPTURE tokens if they were available.

Reply:
@ranj063 yes, it is necessary, we only parse one token each time, here we are using spcm->stream[stream] with direction stream.